includes/compatibility.php

```diff
@@ -4,55 +4,100 @@
-            'check_value' => 'Theme_My_Login'
-        ),
-		array(
-			'file' => 'woocommerce.php',
-			'check_type' => 'constant',
-			'check_value' => 'WC_PLUGIN_FILE'
-		),
-		array(
-			'file' => 'wp-engine.php',
-			'check_type' => 'function',
-			'check_value' => 'wpe_filter_site_url'
-		),
-        array(
-            'file' => 'divi.php',
-            'check_type' => 'constant',
-            'check_value' => 'ET_BUILDER_PLUGIN_DIR'
-        )
-    );
+function pmpro_compatibility_checker() {
+	$compat_checks = [
+		[
+			'file'        => 'siteorigin.php',
+			'check_type'  => 'constant',
+			'check_value' => 'SITEORIGIN_PANELS_VERSION',
+		],
+		[
+			'file'        => 'elementor.php',
+			'check_type'  => 'constant',
+			'check_value' => 'ELEMENTOR_VERSION',
+		],
+		[
+			'file'        => 'beaver-builder.php',
+			'check_type'  => 'constant',
+			'check_value' => 'FL_BUILDER_VERSION',
+		],
+		[
+			'file'        => 'theme-my-login.php',
+			'check_type'  => 'class',
+			'check_value' => 'Theme_My_Login',
+		],
+		[
+			'file'        => 'woocommerce.php',
+			'check_type'  => 'constant',
+			'check_value' => 'WC_PLUGIN_FILE',
+		],
+		[
+			'file'        => 'wp-engine.php',
+			'check_type'  => 'function',
+			'check_value' => 'wpe_filter_site_url',
+		],
+		[
+			'file'        => 'divi.php',
+			'check_type'  => 'constant',
+			'check_value' => 'ET_BUILDER_PLUGIN_DIR',
+		],
+		[
+			'file'                => 'wp-com.php',
+			'check_type'          => 'constant',
+			'check_value'         => 'IS_WPCOM',
+			'check_constant_true' => true,
+		],
+	];
 
-    foreach ( $compat_checks as $key => $value ) {
-        if ( ( $value['check_type'] == 'constant' && defined( $value['check_value'] ) )
-          || ( $value['check_type'] == 'function' && function_exists( $value['check_value'] ) )
-          || ( $value['check_type'] == 'class' && class_exists( $value['check_value'] ) ) ) {
-            include( PMPRO_DIR . '/includes/compatibility/' . $value['file'] ) ;
-        }
+	foreach ( $compat_checks as $value ) {
+		if ( ! pmpro_compatibility_checker_is_requirement_met( $value ) ) {
+			return;
+		}
+
+        include( PMPRO_DIR . '/includes/compatibility/' . $value['file'] ) ;
     }
 }
 add_action( 'plugins_loaded', 'pmpro_compatibility_checker' );
 
+/**
+ * Check whether the requirement is met.
+ *
+ * @since TBD
+ *
+ * @param array $requirement The requirement config (check_type, check_value, check_constant_true).
+ *
+ * @return bool Whether the requirement is met.
+ */
+function pmpro_compatibility_checker_is_requirement_met( $requirement ) {
+	// Make sure we have the keys that we expect.
+	if ( ! isset( $requirement['check_type'], $requirement['check_value'] ) ) {
+		return false;
+	}
+
+	// Check for a constant and maybe check if the constant is true-ish.
+    if ( 'constant' === $requirement['check_type'] ) {
+		return (
+			defined( $requirement['check_value'] )
+			&& (
+				! isset( $requirement['check_constant_true'] )
```

Comment:
```suggestion
				empty( $requirement['check_constant_true'] )
```

I suppose someone might define check_constant_true => false, in which case we also would not want to do the following check